R/covr.R

```diff
@@ -642,7 +642,8 @@ merge_coverage.list <- function(coverage_objs) {
 #   may have been allocated, but not entirely populated.
 #
 clean_coverage_tests <- function(obj) {
-  if (is.null(obj$tests)) return()
+  counter_has_tests_tally <- function(counter) !is.null(counter$tests)
+  if (is.na(Position(counter_has_tests_tally, obj))) return()
```

Comment:
`obj$tests` being `NULL` was not a sufficient way of determining whether counters would have test matrices. Previously, this would mean that matrices would be left with their pre-allocated records. 

Instead, using the existence of the `obj[[key]]$tests` element directly, this is a more reliable way of determining whether tests were captured.